Azure/sonic-buildimage
# [Accton] Support system health monitoring on pddf implemented models

#### Why I did it
To support SystemHealthMonitoring feature on Accton's models (which have implemented PDDF APIs).

#### How I did it
1. Provided a config file, system_health_monitoring_config.json. 
2. Implement the SystemLED part that'd be manipulated by HealthChecker. 
3. Enable thermalctld that makes more HW checks available

#### How to verify it
Use `show system-health detail` and check if all desired checks pass, and confirm if systemLED works reasonably.
```
////// SEAN: Booting, LED_DIAG = GREEN_BLINKING //////

root@sonic:/home/admin# show system-health detail 
System is currently booting...


////// SEAN: verifying with PSU2 unplugged //////
////// SEAN: Failed the health check, LED_DIAG = AMBER //////

root@sonic:/home/admin# show system-health detail
System status summary

  System status LED  amber
  Services:
    Status: OK
  Hardware:
    Status: Not OK
    Reasons: PSU 2 is missing or not available
	     PSU2_FAN1 is broken

System services and devices monitor list

Name                        Status    Type
--------------------------  --------  ----------
sonic                       OK        System
rsyslog                     OK        Process
root-overlay                OK        Filesystem
var-log                     OK        Filesystem
routeCheck                  OK        Program
diskCheck                   OK        Program
container_checker           OK        Program
container_memory_telemetry  OK        Program
PSU2_FAN1                   Not OK    Fan
PSU 2                       Not OK    PSU
PSU1_FAN1                   OK        Fan
PSU 1                       OK        PSU

System services and devices ignore list

Name             Status    Type
---------------  --------  ------
PSU1_FAN1.speed  Ignored   Device
PSU2_FAN1.speed  Ignored   Device
asic             Ignored   Device
psu.voltage      Ignored   Device
psu.temperature  Ignored   Device


////// SEAN: Ignore PSU2 in the config file. //////
////// SEAN: Passed the health check, LED_DIAG = GREEN //////

root@sonic:/home/admin# vi /usr/share/sonic/device/x86_64-accton_as7326_56x-r0/system_health_monitoring_config.json 
root@sonic:/home/admin# show system-health detail
System status summary

  System status LED  green
  Services:
    Status: OK
  Hardware:
    Status: OK

System services and devices monitor list

Name                        Status    Type
--------------------------  --------  ----------
sonic                       OK        System
rsyslog                     OK        Process
root-overlay                OK        Filesystem
var-log                     OK        Filesystem
routeCheck                  OK        Program
diskCheck                   OK        Program
container_checker           OK        Program
container_memory_telemetry  OK        Program
PSU1_FAN1                   OK        Fan
PSU 1                       OK        PSU

System services and devices ignore list

Name             Status    Type
---------------  --------  ------
psu.voltage      Ignored   Device
PSU1_FAN1.speed  Ignored   Device
PSU 2            Ignored   Device
asic             Ignored   Device
PSU2_FAN1        Ignored   Device
psu.temperature  Ignored   Device
PSU2_FAN1.speed  Ignored   Device
```

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog
[Accton] Support system health monitoring on pddf implemented models 

Models: AS9716-32D, AS7326-56X, AS7726-32X, AS7816-64X

#### A picture of a cute animal (not mandatory but encouraged)
